server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -457,8 +457,8 @@ private void executeLocalSearch(Task task, SearchTimeProvider timeProvider, Sear
         if (localIndices == null) {
             return Index.EMPTY_ARRAY; //don't search on any local index (happens when only remote indices were specified)
         }
-        return indexNameExpressionResolver.concreteIndices(clusterState, indicesOptions, true,
-            timeProvider.getAbsoluteStartMillis(), localIndices.indices());
+        return indexNameExpressionResolver.concreteIndices(clusterState, indicesOptions, localIndices,
```

Comment:
I think we should change this `resolveLocalIndices()` method here. I think we should change the `indicesOptions` parameter to `request` (and pass down the request from where it is being invoked).

Then change the `concreteIndices()` method this method invokes. Remove the `options` and `indexExpressions` parameters and add request parameter (of type `IndicesRequest`). Then
in the body of this method derive the indicesOptions and indices from the request parameter.